pyNastran/bdf/cards/material_deps.py

```diff
@@ -259,6 +259,192 @@ def write_card(self, size: int=8, is_double: bool=False) -> str:
+                 b2, b3, a, litk, bigk, expn, tau, adel,
+                 plyuni, tid, hbar, dmax, pe,
+                 user, r01, ds, gic, giic, giiic,
+                 comment=''):
+        MaterialDependence.__init__(self)
+        if comment:
+            self.comment = comment
+
+        self.mid = mid
+        self.ppf_model = ppf_model
+
+        self.y012 = y012
+        self.yc12 = yc12
+        self.ys12 = ys12
+        self.ys22 = ys22
+        self.y11limt = y11limt
+        self.y11limc = y11limc
+        self.ksit = ksit
+        self.ksic = ksic
+        self.b2 = b2
+        self.b3 = b3
+        self.a = a
+        self.litk = litk
+        self.bigk = bigk
+        self.expn = expn
+        self.tau = tau
+        self.adel = adel
+        self.plyuni = plyuni
+        self.tid = tid
+        self.hbar = hbar
+        self.dmax = dmax
+        self.pe = pe
+        self.user = user
+        self.r01 = r01
+        self.ds = ds
+        self.gic = gic
+        self.giic = giic
+        self.giiic = giiic
+
+    @classmethod
+    def add_card(cls, card, comment=''):
+        """
+        Adds a MATDMG card from ``BDF.add_card(...)``
+
+        Parameters
+        ----------
+        card : BDFCard()
+            a BDFCard object
+        comment : str; default=''
+            a comment for the card
+
+        """
+        mid = integer(card, 1, 'mid')
+        ppf_model = string(card, 2, 'PPFMOD')
+
+        y012 = double_or_blank(card, 9, 'Y012')
+        yc12 = double_or_blank(card, 10, 'YC12')
+        ys12 = double(card, 11, 'YS12')
+        ys22 = double(card, 12, 'YS22')
+        y11limt = double(card, 13, 'Y11LIMT')
+        y11limc = double(card, 14, 'Y11LIMC')
+        ksit = double_or_blank(card, 15, 'KSIT')
+        ksic = double_or_blank(card, 16, 'KSIC')
+        b2 = double(card, 17, 'B2')
+        b3 = double(card, 18, 'B3')
+        a = double(card, 19, 'A')
+        litk = double(card, 20, 'LITK')
+        bigk = double(card, 21, 'BIGK')
+        expn = double(card, 22, 'EXPN')
+        tau = double(card, 23, 'TAU')
+        adel = double(card, 24, 'ADEL')
+
+        if ppf_model == "UD":
+            plyuni = integer_or_blank(card, 25, 'PLYUNI')
+            tid = integer_or_blank(card, 26, 'TID')
+            hbar = double(card, 27, 'HBAR')
+            dmax = double(card, 28, 'DMAX')
+            pe = integer_or_blank(card, 29, 'PE')
+
+            user = None
+            r01 = None
+            ds = None
+            gic = None
+            giic = None
+            giiic = None
+        elif ppf_model == "EUD":
+            user = integer(card, 25, 'USER')
+            r01 = double(card, 26, 'R01')
+            hbar = double(card, 27, 'HBAR')
+            dmax = double(card, 28, 'DMAX')
+            ds = double_or_blank(card, 29, 'DS')
+            gic = double(card, 30, 'GIC')
+            giic = double(card, 31, 'GIIC')
+            giiic = double(card, 32, 'GIIIC')
+
```

Comment:
Please add an else: raise RuntimeError(...) to complain about the incorrect ppf_model?